In [ ]:
import requests
import pandas as pd

url = "https://api.nhtsa.gov/complaints/complaintsByVehicle"

params = {
    "make": "jeep",
    "model": "grand cherokee",
    "modelYear": "2012"}

response = requests.get(url, params=params)

if response.status_code == 200:
    data = response.json()
    print(data)
else:
    print("Error: " + str(response.status_code))

{'count': 1578, 'message': 'Results returned successfully', 'results': [{'odiNumber': 11562548, 'manufacturer': 'Chrysler (FCA US, LLC)', 'crash': False, 'fire': False, 'numberOfInjuries': 0, 'numberOfDeaths': 0, 'dateOfIncident': '12/27/2023', 'dateComplaintFiled': '12/29/2023', 'vin': '1C4RJEAG6CC', 'components': 'POWER TRAIN', 'summary': 'The transmission shifts strangely, and it seems like it cannot decide which gear it needs to be in for normal operations. It also causes the vehicle to stall while driving. Also, at times it stays in only one gear, and you cannot even shift gears manually. Intermittently, the vehicle starts out in a higher gear, and it doesn’t correct itself, so acceleration is way below what it needs to be. I have been fortunate to avoid a few crashes due to the stalling.', 'products': [{'type': 'Vehicle', 'productYear': '2012', 'productMake': 'JEEP', 'productModel': 'GRAND CHEROKEE', 'manufacturer': 'Chrysler (FCA US, LLC)'}]}, {'odiNumber': 11560307, 'manufactur

In [ ]:
complaints = pd.DataFrame(data['results'])

In [ ]:
complaints.head()

odiNumber            manufacturer  crash   fire  numberOfInjuries  \
0   11562548  Chrysler (FCA US, LLC)  False  False                 0   
1   11560307  Chrysler (FCA US, LLC)  False  False                 0   
2   11554291  Chrysler (FCA US, LLC)  False  False                 0   
3   11550389  Chrysler (FCA US, LLC)  False  False                 0   
4   11547276  Chrysler (FCA US, LLC)  False  False                 0   

   numberOfDeaths dateOfIncident dateComplaintFiled          vin  \
0               0     12/27/2023         12/29/2023  1C4RJEAG6CC   
1               0     10/15/2023         12/15/2023  1C4RJFAG6CC   
2               0     11/08/2023         11/09/2023  1C4RJFBG5CC   
3               0     08/01/2023         10/17/2023  1C4RJFAG6CC   
4               0     08/14/2023         09/29/2023  1C4RJFBT1CC   

                                      components  \
0                                    POWER TRAIN   
1                              ELECTRICAL SYSTEM   
2  SUSPENSION,POWER TRAIN,FUEL/PROPULSION SYSTEM   
3                              ELECTRICAL SYSTEM   
4                               UNKNOWN OR OTHER   

                                             summary  \
0  The transmission shifts strangely, and it seem...   
1  The contact owns a 2012 Jeep Grand Cherokee. T...   
2  Estaba manejando a 35 millas y de repentes la ...   
3  The contact owns a 2012 Jeep Grand Cherokee. T...   
4  drain water from AC leaking into cabin & flood...   

                                            products  
0  [{'type': 'Vehicle', 'productYear': '2012', 'p...  
1  [{'type': 'Vehicle', 'productYear': '2012', 'p...  
2  [{'type': 'Vehicle', 'productYear': '2012', 'p...  
3  [{'type': 'Vehicle', 'productYear': '2012', 'p...  
4  [{'type': 'Vehicle', 'productYear': '2012', 'p...

In [ ]:
print(complaints['components'].str.split(',\s+', expand=True).stack().value_counts())

ELECTRICAL SYSTEM                                  509
SERVICE BRAKES                                     200
FUEL/PROPULSION SYSTEM                              92
ENGINE                                              78
ELECTRICAL SYSTEM,ENGINE                            62
                                                  ... 
ELECTRICAL SYSTEM,AIR BAGS,ENGINE                    1
AIR BAGS,SERVICE BRAKES,FUEL/PROPULSION SYSTEM       1
POWER TRAIN,ELECTRICAL SYSTEM,UNKNOWN OR OTHER       1
VEHICLE SPEED CONTROL,SERVICE BRAKES,ENGINE          1
ENGINE AND ENGINE COOLING,VEHICLE SPEED CONTROL      1
Length: 175, dtype: int64


Most complaints:
Electrical System 509,
Service Brakes 200,
Fuel Propulsion System 92
